lib/bashly/views/script/header.erb

```diff
@@ -1,3 +1,8 @@
 #!/usr/bin/env bash
 # This script was generated by bashly (https://github.com/DannyBen/bashly)
 # Modifying it manually is not recommended
+
+if [[ "${BASH_VERSINFO:-0}" -lt 4 ]]; then
+  printf "<%= strings[:unsupported_bash_version] -%>\n"
+  exit 1
+fi
```

Comment:
Moved this to the top of the file so Bash 3 can see this and early exit before it gets to any syntax it doesn't understand (it was choking on trying to parse `[[ -v` IIRC). This also raises the question of if any of the other idioms I'm using require a newer version of Bash. Does bashly have a policy on minimum bash version? I briefly investigated running the tests on Bash 4.0 but couldn't immediately find a compiled copy of that which would actually run on my system.

Reply:
Hmm... no, I don't like this. My reasoning:

1. This type of check is exactly where it is supposed to be, logically (in my mind) - in the initialization sequence.
2. Of course, this requires that a) people who develop the bashly gem, havs bash 4, and b) that anything that appears before this check (which is just the 3 lines in the `initialize` function), are not going to fail in old bash versions.

To my knowledge, the current setup - from the standpoint of the generated script users - will always fail gracefully on bash 3. Am I wrong? If not, then I understand that this change is only meant to accommodate bash gem developers, and it is not something we should do, since having bash 4 when developing the bashly gem is a mandatory prerequisite in my mind.

On a more philosophic note - Bash 3 is over a decade long - we cannot encourage bad behavior by operating system vendors, no matter how powerful they are... 

> Does bashly have a policy on minimum bash version?

Yes, [here](https://bashly.dannyb.co/installation/#prerequisites) - bash 4 is a must.

Reply:
> the current setup - from the standpoint of the generated script users - will always fail gracefully on bash 3. Am I wrong?

Yes, this is not correct. If the `_command.sh` files from the user contain any syntax that Bash 3 does not understand, it will fail during parsing rather than gracefully with an explanatory error.

I will extract this change into a separate PR and make a test case for this situation, since it's confusing to have it mixed up with this PR.